### PR TITLE
Parameter Update & NetUse robustness (Closes #8)

### LINKED ISSUE #8: Error when source path contains blank space in xSqlServerSetup
Function netuse in xPDT doesn't quote the path so the command net use failes if source path contains a blank space and comes back with the help text. 

Example:
xSQLServerSetup "SQLServerSetup"
        {
            InstanceName = "MSSQLSERVER"
            SetupCredential = $Credential
            SourcePath = "\\ServerName\Sources\SQL 2012 SP2"
        }

I would recommend using a string builder as it makes quoting much easier.

By the way, same should happen when password contains a $ or an other symbol that is expected to be interpreted by net use.

### PR DESCRIPTION
**Pull request by oWretch**
If UpdateEnabled is 'False', the UpdateSource is still passed to setup.exe. This causes an installation failure if the folder does not exist.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xsqlserver/10)

### CODE REVIEW

**Comment by KarolKaczmarek** [comment]
Thanks @oWretch 
Could you please update readme.md with description of your changes as well?

**Comment by oWretch** [comment]
Updated readme.md

**Comment by TravisEz13** [comment]
Please resolve merge conflicts.  For more information see [GitHowTo - 30. Resolving Conflicts](http://githowto.com/resolving_conflicts )

File: xPDT.psm1 [hunk 1]
```diff
@@ -695,11 +695,11 @@ function NetUse
 
         if ($Ensure -eq "Absent")
         {
-            $cmd = "net.exe use $SourcePath /del"
+            $cmd = "net.exe use `"$SourcePath`" /del"
         }
         else 
         {
-            $cmd = "net.exe use $SourcePath $($Credential.GetNetworkCredential().Password) /user:$($Credential.GetNetworkCredential().Domain)\$($Credential.GetNetworkCredential().UserName)"
+            $cmd = "net.exe use `"$SourcePath`" `"$($Credential.GetNetworkCredential().Password)`" /user:`"$($Credential.GetNetworkCredential().Domain)\$($Credential.GetNetworkCredential().UserName)`""
```
Comment (Arturas-K) [inline xPDT.psm1:702]: I think in latest version this is solved by using parameter arrays #20

**Comment by Arturas-K** [comment]
@oWretch i think you have problem in xSQLServerSetup.psm1 with line endings.
I suggest you open it with Notepad++ and verify that you have LF (unix) line endings and not CRLF (windows)
You can find more info [here](https://github.com/PowerShell/DscResources/blob/TravisEz13-patch-1/StyleGuidelines.md)

**Comment by TravisEz13** [comment]
If there is no update of the pull request within 7 days, we will close the pull request.  After that, if you have an update, feel free to reopen the pull request or a new pull request.

**Comment by TravisEz13** [comment]
Please resolve all merge conflicts.  For more information see [GitHowTo - 30. Resolving Conflicts](http://githowto.com/resolving_conflicts )
[Instructions to setup BC4, my favorite merge tool](http://www.scootersoftware.com/support.php?zz=kb_vcs#gitwindows)

**Comment by TravisEz13** [comment]
If there is no update of the pull request within 7 days, we will close the pull request.  After that, if you have an update, feel free to reopen the pull request or a new pull request.  Please consider closing and re-opening when ready for review.

**Comment by TravisEz13** [comment]
Reviewed 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/10)*

**Pull request closed without merging**